python/hail/api2/table.py

```diff
@@ -444,33 +538,85 @@ def filter(self, expr, keep=True):
     @typecheck_method(exprs=tupleof(oneof(Expression, strlike)),
                       named_exprs=dictof(strlike, anytype))
     def select(self, *exprs, **named_exprs):
-        """Select a subset of columns.
+        """Select existing fields or create new fields by name, dropping the rest.
 
-        **Examples**
+        Examples
+        --------
+        Select a few old columns and compute a new one:
 
-        Assume ``table1`` is a :py:class:`.KeyTable` with three columns: C1, C2 and
-        C3.
+        >>> table_result = table1.select(table1.ID, table1.C1, Y=table1.Z - table1.X)
 
-        Select/drop columns:
+        Notes
+        -----
+        This method creates new row-indexed fields. If a created field shares its name
+        with a global field of the table, the method will fail.
 
-        >>> table_result = table1.select(table1.C1)
+        Note
+        ----
 
-        Reorder the columns:
+        **Using select**
 
-        >>> table_result = table1.select(table1.C3, table1.C1, table1.C2)
+        Select and its sibling methods (:meth:`Table.select_globals`,
+        :meth:`MatrixTable.select_globals`, :meth:`MatrixTable.select_rows`,
```

Comment:
Mixing `Table` and `MatrixTable` here intentionally?

Reply:
yep! Want to show that all these methods are similar.